test/python/transpiler/test_preset_passmanagers.py

```diff
@@ -303,6 +311,24 @@ def counting_callback_func(pass_, dag, time, property_set, count):
         )
         self.assertEqual(gates_in_basis_true_count + 2, consolidate_blocks_count)
 
+    @data(0, 1, 2, 3)
+    def test_layout_registers_preserved(self, optimization_level):
```

Comment:
Do we currently also include the dummy ancilla register in the `Layout`, and if so, can we add a test that checks that that behaviour is maintained too?

Reply:
We do in 2.1.2 because we update the initial layout after the ancilla allocation. In my local testing just now local testing just now this is preserved. I'll add a test case for it here too